share.android.js

```diff
@@ -116,13 +141,31 @@ export default class Share extends Component {
     </Dialog>
   }
 
+  renderSavedDialog = () => {
+    return <Dialog>
+      <Dialog.Title>Saved bookmark</Dialog.Title>
+      <Dialog.Content>
+        <Text style={styles.msg}>{'The article is saved as'}</Text>
+        <Text style={styles.title}>{this.state.article.title}</Text>
+      </Dialog.Content>
+      <Dialog.Actions>
+        <DialogDefaultActions
+          actions={['Cancel', 'Delete', (this.state.article.read ? 'Unread' : 'Read')]}
+          onActionPress={this.handleActionPress}
+        />
+      </Dialog.Actions>
+    </Dialog>
+  }
+
   render() {
     return (
       <ThemeProvider uiTheme={{}}>
         <View style={styles.container}>
-          {(this.state.server && this.state.token)
-          ? this.renderSaveDialog()
-          : this.renderNoAuth()}
+          {(!this.state.server || !this.state.token)
+           ? this.renderNoAuth()
+           : (this.state.article)
+           ? this.renderSavedDialog()
+           : this.renderSaveDialog()}
```

Comment:
I don't care about separating concerns as you do: I think more about readability.